ledger/common/utils/utils.go

```diff
@@ -7,23 +7,22 @@ import (
 	"math"
 )
 
-// IsBitSet returns if the bit at index `idx` in the byte array `b` is set to 1 (big endian)
-// TODO: remove error return to improve performance
-func IsBitSet(b []byte, idx int) (bool, error) {
-	if idx >= len(b)*8 {
-		return false, fmt.Errorf("input (%v) only has %d bits, can't look up bit %d", b, len(b)*8, idx)
-	}
-	return b[idx/8]&(1<<int(7-idx%8)) != 0, nil
-}
-
-// SetBit sets the bit at position i in the byte array b to 1
-// TODO: remove error return to improve performance
-func SetBit(b []byte, i int) error {
-	if i >= len(b)*8 {
-		return fmt.Errorf("input (%v) only has %d bits, can't set bit %d", b, len(b)*8, i)
-	}
-	b[i/8] |= 1 << int(7-i%8)
-	return nil
+// Bit returns the bit at index `idx` in the byte array `b` (big endian)
+//
+// The function assumes b has at least idx bits. The caller must make sure this condition is met.
+func Bit(b []byte, idx int) int {
+	byteValue := int(b[idx>>3])
+	idx &= 7
+	return (byteValue >> (7 - idx)) & 1
+}
```

Comment:
I hope this is the same as above, this is way beyond my understanding, we might add some test for this.

Reply:
It just saves a few cycles compared to before. Also retuning the bit itself rather than a boolean cuts a comparison. 
I'll see if the package `bits` has something similar

Reply:
tests added [here](https://github.com/onflow/flow-go/pull/455/commits/f2e5ca0a2de410282ce802f7a0e5abc967ec5483)